pkg/jobs/jobs.go

```diff
@@ -76,6 +76,50 @@ type Record struct {
 	CreatedBy *CreatedByInfo
 }
 
+// Specification specifies a job, which will be created in a batch with other jobs.
+type Specification struct {
+	jobID  jobspb.JobID
```

Comment:
I'm starting to think maybe we should just stick ID as a field in Record. It'd simplify the code and some function signatures. I don't see any good reason not to. The history here is that the jobs package used to allocate the id for the user. That turned out to be unfortunate so we changed the client to allocate the id. May as well just move the ID into the record.